FirebaseStorageUITests/FUIImageViewCategoryTests.m

```diff
@@ -0,0 +1,188 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import XCTest;
+@import FirebaseStorageUI;
+
+#import "UIImageView+FirebaseStorage.h"
+
+@class MockDownloadTask, MockStorageReference, MockCache;
+
+@interface MockDownloadTask : NSObject <FUIDownloadTask>
+@property (nonatomic, readwrite, getter=isCancelled) BOOL cancelled;
+@end
+
+@implementation MockDownloadTask
+
+- (void)cancel {
+  self.cancelled = YES;
+}
+
+@end
+
+
+@interface MockCache : NSObject <FUIImageCache>
+@property (nonatomic, readonly) NSMutableDictionary *cached;
+@end
+
+@implementation MockCache
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _cached = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
+
+- (void)storeImage:(UIImage *)image forKey:(NSString *)key {
+  self.cached[key] = image;
+}
+
+- (UIImage *)imageFromDiskCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+- (UIImage *)imageFromMemoryCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+@end
+
+
+@interface MockStorageReference : NSObject <FUIStorageReference>
+@property (nonatomic, readwrite) NSString *fullPath;
+@end
+
+@implementation MockStorageReference
```

Comment:
We're basically just reimplementing mocking by adding a bunch of new protocols and classes that junk up the public interface, and adding additional code that we have to maintain. It also raises the barrier to adding new features to mock, since the developer will have to add new protocols/mock classes vs just stubbing them with a mocking framework.

I think it introduces some complexity (developers have to know OCMock), but that is offset by the standardization of the toolset (they should learn it, or they'll know it) as well as the power (partial mocks are a great example of this that we'd have a hard time implementing without rewriting a lot of code).

I'd much rather take on a well known and maintained dependency that developers will be familiar with than try to rebuild out own that solves temporary needs, especially when the barrier to adding the framework is relatively low. Long term, I think we'll be much happier using OCMock to mock other pieces of our system than trying to reimplement everything ourselves.